postgrenosql/src/main/java/com/yahoo/ycsb/postgrenosql/PostgreNoSQLDBClient.java

```diff
@@ -0,0 +1,332 @@
+import java.sql.SQLException;
+import java.util.*;
+
+/**
+ * PostgreNoSQL client for YCSB framework.
+ */
+public class PostgreNoSQLDBClient extends DB {
+  private static final Logger LOG = LoggerFactory.getLogger(PostgreNoSQLDBClient.class);
+
+  /**
+   * Count the number of times initialized to teardown on the last
+   * {@link #cleanup()}.
+   */
+  private static final AtomicInteger INIT_COUNT = new AtomicInteger(0);
+
+  /** The driver to get the connection to postgresql. */
+  private static Driver postgrenosqlDriver;
+
+  /** The connection to the database. */
+  private static Connection connection;
+
+  /** The class to use as the jdbc driver. */
+  public static final String DRIVER_CLASS = "db.driver";
+
+  /** The URL to connect to the database. */
+  public static final String CONNECTION_URL = "postgrenosql.url";
+
+  /** The user name to use to connect to the database. */
+  public static final String CONNECTION_USER = "postgrenosql.user";
+
+  /** The password to use for establishing the connection. */
+  public static final String CONNECTION_PASSWD = "postgrenosql.passwd";
+
+  /** The JDBC connection auto-commit property for the driver. */
+  public static final String JDBC_AUTO_COMMIT = "postgrenosql.autocommit";
+
+  /** The primary key in the user table. */
+  public static final String PRIMARY_KEY = "YCSB_KEY";
+
+  /** The field name prefix in the table. */
+  public static final String COLUMN_NAME = "YCSB_VALUE";
+
+  private static final String DEFAULT_PROP = "";
+
+  private Properties props;
+
+  /** Returns parsed boolean value from the properties if set, otherwise returns defaultVal. */
+  private static boolean getBoolProperty(Properties props, String key, boolean defaultVal) {
+    String valueStr = props.getProperty(key);
+    if (valueStr != null) {
+      return Boolean.parseBoolean(valueStr);
+    }
+    return defaultVal;
+  }
+
+  @Override
+  public void init() throws DBException {
+    INIT_COUNT.incrementAndGet();
+    synchronized (PostgreNoSQLDBClient.class) {
+      if (postgrenosqlDriver != null) {
+        return;
+      }
+
+      props = getProperties();
+      String urls = props.getProperty(CONNECTION_URL, DEFAULT_PROP);
+      String user = props.getProperty(CONNECTION_USER, DEFAULT_PROP);
+      String passwd = props.getProperty(CONNECTION_PASSWD, DEFAULT_PROP);
+      boolean autoCommit = getBoolProperty(props, JDBC_AUTO_COMMIT, true);
+
+      try {
+        Properties tmpProps = new Properties();
+        tmpProps.setProperty("user", user);
+        tmpProps.setProperty("password", passwd);
+
+        postgrenosqlDriver = new Driver();
+        connection = postgrenosqlDriver.connect(urls, tmpProps);
+        connection.setAutoCommit(autoCommit);
+      } catch (Exception e) {
+        LOG.error("Error during initialization: " + e);
+      }
+    }
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    if (INIT_COUNT.decrementAndGet() == 0) {
+      postgrenosqlDriver = null;
```

Comment:
need to call `close` on the Connection instance as a part of cleanup. if autocommit wasn't configured, you need to expressly call `commit` prior to `close` as well.